apis/postgresql/v1alpha1/provider_types.go

```diff
@@ -33,7 +33,8 @@ type ProviderConfigSpec struct {
 	// Defines the SSL mode used to set up a connection to the provided
 	// PostgreSQL instance
 	// +kubebuilder:validation:Enum=disable;require;verify-ca;verify-full
-	// +optional
+	// +kubebuilder:default=verify-full
+	// +kubebuilder:validation:Required
 	SSLMode string `json:"sslMode"`
```

Comment:
I believe we should set it as optional (with omit-empty) since we have a default, e.g. 

https://github.com/crossplane/crossplane/blob/7d01448f8a52e0f3a13add4fc0f81028fcd13400/apis/pkg/v1/package_types.go#L31

Reply:
Thanks @turkenh, converted it to an optional field.

Reply:
Should this be `*string`? (Kubernetes API guidelines require nil-able optional fields.)